src/index.ts

```diff
@@ -34,73 +38,31 @@ export async function execute(filenames: string[], options: Partial<ExecutionOpt
     const executionOptions = Object.assign(defaultExecutionOptions, options);
 
     let manifest = await PP.getManifest(executionOptions.root);
+    let componentDefinitions = await getComponentDefinitionMap(executionOptions.root);
```

Comment:
Does the interpreter not need any other execution options?

Reply:
The component parsing? Ultimately it just needs to know the root of either where it's executing or where you tell the CLI to execute. It will scan that space for components. It doesn't need any of the I/O or other options.